### PR TITLE
Update example docstring of raw_ops.UniqueV2

### PR DESCRIPTION
**Pull request by sachinprasadhs**
Updated example docstring with the working code.

### CODE REVIEW

File: tensorflow/core/api_def/base_api/api_def_UniqueV2.pbtxt [hunk 1]
```diff
@@ -40,36 +40,39 @@ In other words, for an `1-D` tensor `x` with `axis = None:
...
 
-```
-# tensor 'x' is [1, 1, 2, 4, 4, 4, 7, 8, 8]
-y, idx = unique(x)
-y ==> [1, 2, 4, 7, 8]
-idx ==> [0, 0, 1, 2, 2, 2, 3, 4, 4]
-```
+>>> x = tf.constant([1, 1, 2, 4, 4, 4, 7, 8, 8])
+>>> y, idx = tf.raw_ops.UniqueV2(x=x, axis=[0])
+>>> print(y)
```
Comment (mihaimaruseac) [inline tensorflow/core/api_def/base_api/api_def_UniqueV2.pbtxt:45]: Where is the output of this?

File: tensorflow/core/api_def/base_api/api_def_UniqueV2.pbtxt [hunk 2]
```diff
@@ -40,36 +40,39 @@ In other words, for an `1-D` tensor `x` with `axis = None:
...
-y, idx = unique(x, axis=0)
-y ==> [[1, 0, 0],
-       [2, 0, 0]]
-idx ==> [0, 0, 1]
-```
+>>> x = tf.constant([[1, 0, 0],
+...                 [1, 0, 0],
+...                 [2, 0, 0]])
+>>> y, idx = tf.raw_ops.UniqueV2(x=x, axis=[0])
+>>> print(y)
```
Comment (mihaimaruseac) [inline tensorflow/core/api_def/base_api/api_def_UniqueV2.pbtxt:56]: Same here.

File: tensorflow/core/api_def/base_api/api_def_UniqueV2.pbtxt [hunk 3]
```diff
@@ -40,36 +40,39 @@ In other words, for an `1-D` tensor `x` with `axis = None:
...
-y ==> [[1, 0],
-       [1, 0],
-       [2, 0]]
-idx ==> [0, 1, 1]
-```
+>>> x = tf.constant([[1, 0, 0],
+...                [1, 0, 0],
+...                [2, 0, 0]])
+>>> y, idx = tf.raw_ops.UniqueV2(x=x, axis=[1])
+>>> print(y)
```
Comment (mihaimaruseac) [inline tensorflow/core/api_def/base_api/api_def_UniqueV2.pbtxt:69]: And here

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
Please make sure you run at the least the docstring test and it passes, instead of just doing mechanical changes.

**Comment by mihaimaruseac** [comment]
You should move the Python type doctests to a Python file, not to a textual protobuf that should be valid across multiple languages

**Comment by sachinprasadhs** [comment]
I guess the raw_ops does not have python file, and the example mentioned here https://www.tensorflow.org/api_docs/python/tf/raw_ops/UniqueV2 in the document throws an exception.

**Comment by gbaned** [comment]
Hi @jsmeredith Can you please review this PR ? Thank you!

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
Please move Python doctests to Python files.

**Comment by sachinprasadhs** [comment]
@mihaimaruseac , Could you please review the changes, I have modified according to your suggestion.
> You should move the Python type doctests to a Python file, not to a textual protobuf that should be valid across multiple languages

**Comment by sachinprasadhs** [comment]
The python file I could find for Tensorflow unique is here https://github.com/tensorflow/tensorflow/blob/v2.11.0/tensorflow/python/ops/array_ops.py#L2022-L2064, but it is still not migrated to UniqueV2.
But the above changes are made to the tf.raw_ops.UniqueV2 for which I could not find python file, the current example is mentioned in pbtxt file. 
> Please move Python doctests to Python files.

**Comment by sachinprasadhs** [comment]
The python file I could find for Tensorflow unique is here https://github.com/tensorflow/tensorflow/blob/v2.11.0/tensorflow/python/ops/array_ops.py#L2022-L2064, but it is still not migrated to UniqueV2.
But the above changes are made to the tf.raw_ops.UniqueV2 for which I could not find python file, the current example is mentioned in pbtxt file. 
> Please move Python doctests to Python files.

**Comment by mihaimaruseac** [comment]
You can attach the docstring to a new symbol in the Python file and remove it from the proto

**Comment by mihaimaruseac** [comment]
See b2755d85dbf99a6d3cf6ffd92e358377e3c46652 for an example

**Comment by mihaimaruseac** [comment]
Hi @sachinprasadhs , any update here?

File: tensorflow/core/api_def/base_api/api_def_UniqueV2.pbtxt [hunk 4]
```diff
@@ -41,19 +41,19 @@ In other words, for an `1-D` tensor `x` with `axis = None:
...
 ```
 
 For an `2-D` tensor `x` with `axis = 0`:
 
 ```
-# tensor 'x' is [[1, 0, 0],
-#                [1, 0, 0],
-#                [2, 0, 0]]
-y, idx = unique(x, axis=0)
+x = tf.constant([[1, 0, 0],
```
Comment (ishark) [inline tensorflow/core/api_def/base_api/api_def_UniqueV2.pbtxt:53]: While at it, could you please add the example as a testable docstring?

E.g.
>>>  x = tf.constant[1, 0, 0],
...                               [1, 0, 0],
...                               [2, 0, 0]]
>>> y, idx = tf.raw_ops.UniqueV2(x=x, axis=[0])
>>> print(y)
tf.Tensor(
[[1 0 0]
 [2 0 0]], shape=(2, 3), dtype=int32)
>>> print(idx)
tf.Tensor([0 0 1], shape=(3,), dtype=int32)



Here are a couple of examples: 
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/api_def/base_api/api_def_Tile.pbtxt
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/api_def/base_api/api_def_Sign.pbtxt

**Review by ishark — COMMENTED** [review]
Thank you for updating the documents!

File: tensorflow/core/api_def/base_api/api_def_UniqueV2.pbtxt (hunk 4 above)
Comment (mihaimaruseac) [reply tensorflow/core/api_def/base_api/api_def_UniqueV2.pbtxt:53]: Please don't use docstrings in base_api/pbtxt. To make them testable docstrings you have to extract them to Python files first.

Comment (mihaimaruseac) [reply tensorflow/core/api_def/base_api/api_def_UniqueV2.pbtxt:53]: I added an example at https://github.com/tensorflow/tensorflow/pull/57873#issuecomment-1331485417

Comment (ishark) [reply tensorflow/core/api_def/base_api/api_def_UniqueV2.pbtxt:53]: Ah, please ignore my comment about testable doc string.
Thanks for pointing it out Mihai.

**Comment by sachinprasadhs** [comment]
@mihaimaruseac , Apologies for the delayed response.
One confusion here. 
The Docstring in pbtxt which I'm trying to update is https://www.tensorflow.org/api_docs/python/tf/raw_ops/UniqueV2, which is having issues with the `axis` as the argument.
The Python file where you are suggesting to update is https://www.tensorflow.org/api_docs/python/tf/unique which is still v1 behavior and does not have axis as the argument.
If I disable the visibility of pbtxt for `tf.raw_ops.UniqueV2` and update the docstring of `tf.unique` does not serve the purpose of this PR. 
Could you please correct me if my understanding is wrong here. Thanks

**Comment by mihaimaruseac** [comment]
Please see the changes to the python file in b2755d85dbf99a6d3cf6ffd92e358377e3c46652.

You need to create a new Python function to which you attach the docstring. Since it is now in Python, it acts as a testable docstring.

**Comment by sachinprasadhs** [comment]
@mihaimaruseac , Will there be any conflicts with the new changes? Since the existing `gen_array_ops.py` uses `UniqueV2 = tf_export("raw_ops.UniqueV2")(_ops.to_raw_op(unique_v2))`

**Comment by sachinprasadhs** [comment]
How do I do that, since the existing one is in genfile.

**Comment by mihaimaruseac** [comment]
You have to mark the pbtxt symbol `HIDDEN`.

**Comment by sachinprasadhs** [comment]
I have already made it hidden.

**Comment by mihaimaruseac** [comment]
In this case, the API generation process will no longer generate the old `tf_export`. But you have to run it

**Review by mihaimaruseac — APPROVED** [review]
(pending API generation, tests might fail otherwise).

**Comment by sachinprasadhs** [comment]
I guess it is still generating old `tf_export` and failing with the error `__main__.SymbolExposedTwiceError: Trying to export multiple symbols with same name: raw_ops.UniqueV2`

**Review by gbaned — APPROVED** [review]

**Pull request closed without merging**